### PR TITLE
Fix/motor tests

### PR DESCRIPTION
**Pull request by Oozyx**
## Changes

- Add test suite with test setup and teardown
- Add registry tests to test suite
- Fix concurrent map writes error with Mutex

## Checklist

- [ ] Unit tests
- [ ] Documentation

## References *(optional)*

Fixes 
Connects 

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: pkg/crypto/mpc/wallet.go [hunk 1]
```diff
@@ -41,7 +42,9 @@ func GenerateWallet(cb common.MotorCallback, options ...WalletOption) (*Wallet,
 			if err != nil {
 				return
 			}
+			w.ConfigsLock.Lock()
 			w.Configs[conf.ID] = conf
+			w.ConfigsLock.Unlock()
```
Comment (joshLong145) [inline pkg/crypto/mpc/wallet.go:47]: thank you :)

**Review by joshLong145 — APPROVED** [review]
Looking good. Approved

**Review by mcjcloud — APPROVED** [review]
lgtm

**Comment by ntindle** [comment]
When this runs, please check the GitHub actions output for Build/Test. The tests failing currently do not stop the pipeline. Make sure the tests don’t fail there

**Pull request merged**